index.js

```diff
@@ -63,41 +77,52 @@ function fastifyCors (fastify, opts, next) {
         return next(new Error('Invalid CORS origin option'))
       }
 
-      // Enable preflight
-      req.corsPreflightEnabled = true
-
-      reply.header('Access-Control-Allow-Origin',
-        getAccessControlAllowOriginHeader(req.headers.origin, resolvedOriginOption))
-
-      if (credentials) {
-        reply.header('Access-Control-Allow-Credentials', 'true')
-      }
-
-      if (exposedHeaders !== null) {
-        reply.header(
-          'Access-Control-Expose-Headers',
-          Array.isArray(exposedHeaders) ? exposedHeaders.join(', ') : exposedHeaders
-        )
+      addCorsHeaders(req, reply, resolvedOriginOption)
+
+      if (req.raw.method === 'OPTIONS' && preflight === true) {
+        req.corsPreflightEnabled = true
```

Comment:
i'd move this after the strictPreflight condition exclusion for more readability